File: c/src/pje_0009.c
#include <stdio.h>

int main() {
	int a, b, c;
	long long product = 0;
	for (a = 998; a >= 2 && product == 0; a--) {
		for (b = a; b >= 2 && product == 0; b--) {
			for (c = b; c >= 2 && product == 0; c--) {
				if (a+b+c == 1000) {
					if (a > b && a > c) {
						if (b*b+c*c == a*a)
							product = a*b*c;
					}
					else if (b > a && b > c) {
						if (a*a+c*c == b*b)
							product = a*b*c;
					}
					else if (c > a && c > b) {
						if (a*a+b*b == c*c)
							product = a*b*c;
					}
				}
			}
		}
	}
	printf("%lld\n", product);
	return 0;
}

File: python/src/pje_0009.py
a, b, c = 1, 2, 3
a_t, b_t, c_t = 1, 1, 1
done = 0
product = 0

while c < 1000 and done == 0:
	b = 2
	while b < c and done == 0:
		a = 1
		while a < b and done == 0:
			if a + b + c == 1000:
				test = a**2 + b**2
				if test == c**2:
					done = 1
					product = a * b * c
					a_t, b_t, c_t = a, b, c
			a = a + 1
		b = b + 1
	c = c + 1

print "%d" % (product)
